src/main/java/org/GehtSoftHWByAziz/HW4/CustomHashMap.java

```diff
@@ -0,0 +1,195 @@
+package org.GehtSoftHWByAziz.HW4;
+
+import java.util.*;
+
+public class CustomHashMap <K,V>implements Map<K,V> {
+  final private int INIT_TABLE_SIZE = 10;
+  final private int GROW_COEFFICIENT = 2;
+  private  Collection <V>values;
+  private Set<K> keys;
```

Comment:
HashSet is in fact a hashmap with null values (only keys are present)
so, you're using hashmap in a hashmap implementation... hm 🙂